content/courses/native-onchain-development/serialize-instruction-data-frontend.md

```diff
@@ -354,30 +354,53 @@ Now that we have the buffer layout set up, let’s create a method in `Movie`
 called `serialize()` that will return a `Buffer` with a `Movie` object’s
 properties encoded into the appropriate layout.
 
+Instead of allocating a fixed buffer size, we'll calculate the size dynamically
+using known constants for the space required by each field in the `Movie`
+object. Specifically, we'll use `INIT_SPACE` (to account for string length
+metadata) and `ANCHOR_DISCRIMINATOR` (to account for the 8-byte discriminator
+used by Anchor).
+
 ```typescript
-import * as borsh from '@coral-xyz/borsh'
+import * as borsh from "@coral-xyz/borsh";
+
+// Constants for size calculations
+const ANCHOR_DISCRIMINATOR = 8; // Typically 8 bytes for the Anchor discriminator
+const STRING_LENGTH_SPACE = 4; // 4 bytes to store the length of each string (used by Borsh)
+const INIT_SPACE = STRING_LENGTH_SPACE * 2; // Two strings: 'title' and 'description'
```

Comment:
This is confusing. There is something called INIT_SPACE in Anchor which is the space of an entire Rust struct. This is just the space required to store the size of two strings.

Reply:
fixed ✅